src/idl_gen_ts.cpp

```diff
@@ -464,6 +480,22 @@ class TsGenerator : public BaseGenerator {
       case BASE_TYPE_BOOL: return allowNull ? "boolean|null" : "boolean";
       case BASE_TYPE_LONG:
       case BASE_TYPE_ULONG: return allowNull ? "bigint|null" : "bigint";
+      case BASE_TYPE_ARRAY: {
+        if (type.element == BASE_TYPE_LONG || type.element == BASE_TYPE_ULONG) {
+          return allowNull ? "bigint[]|null" : "bigint[]";
```

Comment:
return "bigint[]" + (allowNull ? |"null" : "");

reads better. Same with line 488.

Reply:
The compiler didn't like concatting the two constant strings as it would make the type information unclear so had to do a little more formatting / naming within the block, I think it is even better now. It also improved the formatting of the generated code. c716af815b3b6b5dd3017090a5bdd0ca434d4470